### PR TITLE
(6_1_X) [TIMOB-24796] iOS: Hybrid CommonJS and Native modules don't handl…

### PR DESCRIPTION
**Pull request by sgtcoolguy**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24796

**Optional Description:**
back port of #9131

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
QE/FR steps: https://jira.appcelerator.org/browse/TIMOB-24796?focusedCommentId=421562&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-421562

**Review by hansemannn — APPROVED** [review]
CR looks fine, let's go for it!

File: iphone/Classes/TiModule.m [hunk 1]
```diff
@@ -205,7 +205,15 @@ -(NSData*)loadModuleAsset:(NSString*)fromPath
 {
     [self loadAssets];
     if ([moduleAssets respondsToSelector:@selector(resolveModuleAsset:)]) {
-        return [moduleAssets performSelector:@selector(resolveModuleAsset:) withObject:fromPath];
+       // Look up the file by it's original name. We preserve the filename in our Node-js based module build
+        NSData* data = [moduleAssets performSelector:@selector(resolveModuleAsset:) withObject:fromPath];
+        if (data != null) {
```
Comment (hansemannn) [inline iphone/Classes/TiModule.m]: Use `nil` please

**Review by hansemannn — CHANGES_REQUESTED** [review]

**Review by hansemannn — APPROVED** [review]

**Review by mukherjee2 — APPROVED** [review]
Passed FR! With this environment:
Node Version: 6.10.3
NPM Version: 3.10.10
Mac OS: 10.12.4
Appc CLI: 6.2.2
Appc CLI NPM: 4.2.9
Titanium SDK version: 6.1.1 pr/9132
Appcelerator Studio, build: 4.9.0.201705302345-39
Xcode 8.3.2

I followed the instructions in the ticket to build the module. The module was successfully built. Then I used the module in the demo app. App worked as expected, and no errors were seen.

**Pull request merged**